src/types.rs

```diff
@@ -434,9 +435,56 @@ impl ObjectFeatures {
     }
 }
 
+/// Information about a Debug Information File in the [`CompleteObjectInfo`].
+///
+/// All DIFs are backed by an [`ObjectFile`](crate::actors::objects::ObjectFile).  But we
+/// may not have been able to get hold of this object file.  We still want to describe the
+/// relevant DIF however.
+///
+/// Currently has no [`ObjectId`] attached and the parent container is expected to know
+/// which ID this DIF info was for.
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+pub struct DifInfo {
+    /// The ID of the object source where this DIF was expected to be found.
+    ///
+    /// This refers back to the IDs of sources in the symbolication requests, as well as any
+    /// globally configured sources from symbolicator's configuration.
+    ///
+    /// Generally this is a short readable string.
+    pub source_id: SourceId,
+    /// The location of this DIF on the object source.
+    ///
+    /// This will generally be some sort of path name or key, if you know the type of object
+    /// source this may give you an idea of where this DIF was expected to be found.
+    pub source_location: SourceLocation,
+    /// The status of this DIF.
+    pub status: DifStatus,
+    /// The features provided by this DIF.
+    // TODO(flub): document better + hide when we don't know.
+    pub features: ObjectFeatures,
```

Comment:
Let's give this a `#[serde(skip_serializing_if = "...")]` to ensure we write smaller responses.